### PR TITLE
Add skip_mismatch argument to load_weights.

### PR DESCRIPTION
**Pull request by hgaiser**
This PR is analogous to https://github.com/keras-team/keras/pull/8462 in Keras.

With it, you can load weights into models that have a mismatch, but use the same names. A practical application would be for transfer learning, where the number of classes influences the size of the weights. It can still be useful to load the weights that are not changed in shape to provide a better starting point.

Note: I don't have a TPU, I am relying on the unit tests to tell me if it is correctly implemented or not.

### CODE REVIEW

File: tensorflow/python/keras/engine/network.py [hunk 1]
```diff
@@ -1368,7 +1371,7 @@ def load_weights(self, filepath, by_name=False):
       if 'layer_names' not in f.attrs and 'model_weights' in f:
         f = f['model_weights']
       if by_name:
-        saving.load_weights_from_hdf5_group_by_name(f, self.layers)
+        saving.load_weights_from_hdf5_group_by_name(f, self.layers, skip_mismatch=skip_mismatch)
```
Comment (frankchn) [inline tensorflow/python/keras/engine/network.py]: Line length max 80 characters.

File: tensorflow/python/keras/saving/hdf5_format_test.py [hunk 2]
```diff
@@ -335,6 +336,46 @@ def test_sequential_weight_loading_group_name_with_incorrect_length(self):
...
+      model.add(keras.layers.Dense(num_hidden, use_bias=False,
+                                   input_dim=input_dim, name='d1'))
+      model.add(keras.layers.Dense(num_classes + 1, name='d2')) #  This shape is modified.
+      model.compile(loss=keras.losses.MSE,
+                    optimizer=keras.optimizers.RMSprop(lr=0.0001),
+                    metrics=[keras.metrics.categorical_accuracy])
+    with warnings.catch_warnings(record=True) as w:
+      warnings.simplefilter("always")
+
+      hdf5_format.load_weights_from_hdf5_group_by_name(f_model, model.layers, skip_mismatch=True)
```
Comment (frankchn) [inline tensorflow/python/keras/saving/hdf5_format_test.py]: Line length here and on line 340 and 367 (also maybe other places)

**Review by frankchn — CHANGES_REQUESTED** [review]
Minor line length issues, otherwise OK

File: tensorflow/python/keras/saving/hdf5_format_test.py (hunk 2 above)
Comment (hgaiser) [reply tensorflow/python/keras/saving/hdf5_format_test.py]: Ah line lengths, fun :)

Do you have a proposal for how to modify line 340? I choose the same name as the unit test it is based on and added `_skip_mismatch`, but it makes the name already be longer than 80 characters.

Comment (frankchn) [reply tensorflow/python/keras/saving/hdf5_format_test.py]: I think the usual way to do this is

```
      hdf5_format.load_weights_from_hdf5_group_by_name(
          f_model, model.layers, skip_mismatch=True)
```

File: tensorflow/python/keras/saving/hdf5_format_test.py [hunk 3]
```diff
@@ -335,6 +336,50 @@ def test_sequential_weight_loading_group_name_with_incorrect_length(self):
                                  r'element\(s\)\.'):
       hdf5_format.load_weights_from_hdf5_group_by_name(f_model, model.layers)
 
+  @test_util.run_in_graph_and_eager_modes
+  def test_sequential_weight_loading_group_name_with_incorrect_length_with_skip_mismatch(self):
```
Comment (frankchn) [inline tensorflow/python/keras/saving/hdf5_format_test.py]: Let's just add a `# pylint: disable=line-too-long` to the end here.

**Comment by gbaned** [comment]
@hgaiser Did you get a chance to look on reviewer comments? Please let us know on the update. Thanks!

**Comment by hgaiser** [comment]
Hi, I'm currently on vacation but I will look at it in two weeks.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by hgaiser** [comment]
@tensorflowbutler I will pick it up soon, hopefully.

**Review by frankchn — APPROVED** [review]

**Comment by gbaned** [comment]
@hgaiser Could you please check build failures and resolve the conflicts? Thanks!

**Comment by hgaiser** [comment]
> @hgaiser Could you please check build failures and resolve the conflicts? Thanks!

I believe this was made redundant by https://github.com/tensorflow/tensorflow/commit/c52ee536c045186476450cf76461d47fb2667e7d#diff-4ee308ea180d49ae81691348531a2b6d . Closing this.
